### PR TITLE
Chore/data — Add telemetry events for SSO

### PR DESCRIPTION
**Pull request by Aurelsicoko**
### What does it do?

It adds telemetry events for SSO, Audit Logs and Releases.

### Why is it needed?

We need it to offer better support for the users who configured SSO, used Audit Logs or Releases.

### CODE REVIEW

File: packages/core/admin/ee/server/src/controllers/audit-logs.ts [hunk 1]
```diff
@@ -20,5 +20,7 @@ export default {
     const body = await auditLogs.findOne(id);
 
     ctx.body = body;
+
+    strapi.telemetry.send('didWatchAnAuditLog');
```
Comment (alexandrebodin) [inline packages/core/admin/ee/server/src/controllers/audit-logs.ts:24]: If you want something more precise you might want to move that in the frontend so you only send the event if a user actually open a view. here you will count any api call made (might not by 1:1)

File: packages/core/admin/server/src/services/metrics.ts [hunk 2]
```diff
@@ -19,12 +22,37 @@ const sendDidChangeInterfaceLanguage = async () => {
   strapi.telemetry.send('didChangeInterfaceLanguage', { userProperties: { languagesInUse } });
 };
 
+const getProvidersListSSO = async () => {
```
Comment (alexandrebodin) [inline packages/core/admin/server/src/services/metrics.ts:25]:
```suggestion
const getSSOProvidersList = async () => {
```

File: packages/core/admin/server/src/services/metrics.ts [hunk 3]
```diff
@@ -19,12 +22,37 @@ const sendDidChangeInterfaceLanguage = async () => {
...
+  const { providerRegistry } = strapi.admin.services.passport;
+
+  return providerRegistry.getAll().map(({ uid }) => uid);
+}
+
 const sendUpdateProjectInformation = async () => {
```
Comment (alexandrebodin) [inline packages/core/admin/server/src/services/metrics.ts:31]: I think you need to move that logic into the ee/ code instead

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

File: packages/core/admin/ee/server/src/controllers/audit-logs.ts (hunk 1 above)
Comment (Aurelsicoko) [reply packages/core/admin/ee/server/src/controllers/audit-logs.ts:24]: If an API call is made here, it means the audit log is watched outside the Admin Panel, which still proves a usage of the feature, which is my goal here.

File: packages/core/admin/server/src/services/metrics.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import type { Strapi } from '@strapi/types';
```
Comment (alexandrebodin) [inline packages/core/admin/server/src/services/metrics.ts:1]: You can revort the changes to this file now

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
One last cleanup but looks good

**Review by alexandrebodin — APPROVED** [review]

File: packages/core/content-releases/server/src/services/release.ts [hunk 5]
```diff
@@ -619,6 +625,8 @@ const createReleaseService = ({ strapi }: { strapi: LoadedStrapi }) => ({
       },
     });
 
+    strapi.telemetry.send('didPublishContentRelease');
```
Comment (Feranchz) [inline packages/core/content-releases/server/src/services/release.ts:628]: We already have this event being tracked (as well as for creating a release) in the front with different names "didCreateRelease" and "didPublishRelease"

Adding this with a different name in the backend is on purpose? (To compare vs the front one) Or should we keep only one?

Comment (Aurelsicoko) [reply packages/core/content-releases/server/src/services/release.ts:628]: @Feranchz Did you test them? I confirm that we don't receive any events named like this on our systems. In our case, they may be implemented, but they don't work.

**Pull request merged**